docs/requirements.txt

```diff
@@ -2,3 +2,7 @@ sphinx
 sphinx_rtd_theme
 sphinxcontrib-seqdiag
 numpydoc
+
+cairocffi >= 0.9.0
+cffi >= 1.1.0
+xcffib >= 0.5.0
```

Comment:
IIRC we don't have this here because we have dev.sh

Reply:
(and because they need to be installed in a certain order, which a requirements file or pip itself can not guarantee)

Reply:
but Travis doesn't use `dev.sh` or I miss something?

Reply:
Mhhh, well, it uses what you want it to use? Right now it uses tox, which itself handles its dependencies with `deps` in each environment definition. But what is the relation between this PR and Travis?

Reply:
yes, tox. But Travis builds failed without that :)

Reply:
Do you have a link to a failed Travis build due to this?

Reply:
It is hard to find... Easier to create new build without these lines. Will give you soon. (maybe it is not needed anymore, because I changed a lot)

Reply:
fail https://travis-ci.org/github/qtile/qtile/jobs/680551947

Reply:
So yeah, I guess it should work to just add cairocffi as deps in the tox environment for docs?

Reply:
The begining of this history is with some lessons :) I've started building documentation as described in docs/BUILDING. So created that environment, got issues with sphinx doc building because it self searched virtenv and found warnings. Then this cairocffi issue. Added all dependencies and got documentation. I didn't ues tox locally (or I don't know that I use it :D), so these dependencies will not brake something (now)

Reply:
I'm really sorry, this whole discussion is due to a misreading and a bad memory. I'll do a last suggestion and then have some rest, because apparently I need some. :)